superset/databases/schemas.py

```diff
@@ -272,6 +270,12 @@ def build_sqlalchemy_uri(
             engine_spec = engine_specs[engine]
 
             if hasattr(engine_spec, "build_sqlalchemy_uri"):
+                serialized_encrypted_extra = data.get("encrypted_extra", "{}")
+                try:
+                    encrypted_extra = json.loads(serialized_encrypted_extra)
+                except json.decoder.JSONDecodeError:
+                    encrypted_extra = {}
```

Comment:
I moved this closer to where is needed, to prevent us from running this if not necessary.